Sources/MapboxNavigation/LaneView.swift

```diff
@@ -1,342 +1,337 @@
-            secondaryIndication = .slightRight
-        } else {
-            secondaryIndication = nil
+        let dominantSide = lane.dominantSide(maneuverDirection: maneuverDirection, drivingSide: drivingSide)
+        let turnClassifications = Set(laneIndications.map {
+            TurnClassification(laneIndication: $0, dominantSide: dominantSide, drivingSide: drivingSide)
+        })
+        let favoredTurnClassification = favoredIndication.map {
+            TurnClassification(laneIndication: $0, dominantSide: dominantSide, drivingSide: drivingSide)
         }
-        
-        if let secondaryIndication = secondaryIndication {
-            indications.remove(secondaryIndication)
-            
-            // Some triple-use configurations are supported.
-            let tertiaryIndication: LaneIndication?
-            if !primaryIndication.isSubset(of: [.straightAhead, .uTurn]) && !secondaryIndication.isSubset(of: [.straightAhead, .uTurn]) &&
-                indications.contains(.straightAhead) {
-                tertiaryIndication = .straightAhead
-            } else if (primaryIndication == .straightAhead && !secondaryIndication.isDisjoint(with: .rights) ||
-                        !primaryIndication.isDisjoint(with: .rights) && secondaryIndication == .straightAhead) &&
-                        !indications.isDisjoint(with: .lefts) {
-                tertiaryIndication = .left
-            } else if (primaryIndication == .straightAhead && !secondaryIndication.isDisjoint(with: .lefts) ||
-                        !primaryIndication.isDisjoint(with: .lefts) && secondaryIndication == .straightAhead) &&
-                        !indications.isDisjoint(with: .rights) {
-                tertiaryIndication = .right
-            } else if (primaryIndication == .straightAhead && !secondaryIndication.isSubset(of: [.straightAhead, .uTurn]) ||
-                        !primaryIndication.isSubset(of: [.straightAhead, .uTurn]) && secondaryIndication == .straightAhead) &&
-                        indications == .uTurn {
-                tertiaryIndication = .uTurn
-            } else {
-                tertiaryIndication = nil
-            }
-            return Ranking(primary: primaryIndication, secondary: secondaryIndication, tertiary: tertiaryIndication)
-        } else {
-            // No secondary indication to draw.
-            return Ranking(primary: primaryIndication, secondary: nil, tertiary: nil)
+        guard let method = styleKitMethod(turnClassifications: turnClassifications, favoredTurnClassification: favoredTurnClassification) ??
+                favoredTurnClassification.map({ styleKitMethod(turnClassifications: [$0], favoredTurnClassification: $0) }) ??
+                styleKitMethod(turnClassifications: [.straightAhead], favoredTurnClassification: nil) else {
+            preconditionFailure("No StyleKit method for straight ahead.")
         }
+        return method
```

Comment:
This is the crux of the change, the top level of the algorithm that was implemented in Swift.